node/src/components/console/tasks.rs

```diff
@@ -0,0 +1,335 @@
+            }
+            Outcome::Failure { reason } => {
+                write!(f, "ERR  {}", reason)
+            }
+        }
+    }
+}
+
+/// Configuration for a connection console session.
+#[derive(Copy, Clone, Debug, Default, Serialize)]
+struct Session {
+    /// Whether or not to suppress the operation outcome.
+    quiet: bool,
+    /// Output format to send to client.
+    output: OutputFormat,
+}
+
+impl Display for Session {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        Debug::fmt(self, f)
+    }
+}
+
+impl Session {
+    /// Creates a serializer for an `EraDump`.
+    fn create_era_dump_serializer(&self) -> fn(&EraDump<'_>) -> Result<Vec<u8>, Cow<'static, str>> {
+        // TODO: This function could probably be a generic serialization function for any `T`, but
+        // the conversion is tricky due to the lifetime arguments on `EraDump` and has not been done
+        // yet.
+        match self.output {
+            OutputFormat::Interactive => |data: &EraDump| {
+                let mut buf = data.to_string().into_bytes();
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Json => |data: &EraDump| {
+                let mut buf = serde_json::to_vec(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as JSON: {}", err))
+                })?;
+                buf.push(b'\n');
+                Ok(buf)
+            },
+            OutputFormat::Bincode => |data: &EraDump| {
+                bincode::serialize(&data).map_err(|err| {
+                    Cow::Owned(format!("failed to serialize era dump as bincode: {}", err))
+                })
+            },
+        }
+    }
+
+    /// Processes a single command line sent from a client.
+    async fn process_line<REv>(
+        &mut self,
+        effect_builder: EffectBuilder<REv>,
+        writer: &mut OwnedWriteHalf,
+        line: &str,
+    ) -> io::Result<()>
+    where
+        REv: From<DumpConsensusStateRequest> + Send,
+    {
+        debug!(%line, "line received");
+        match Command::from_line(line) {
+            Ok(ref cmd) => {
+                info!(?cmd, "processing command");
+                match cmd.action {
+                    Action::Session => {
+                        self.send_outcome(writer, &Outcome::success("showing session info"))
+                            .await?;
+                        self.send_to_client(writer, &self).await?;
+                    }
+                    Action::Set { quiet, output } => {
+                        let mut changed = false;
+
+                        if let Some(quiet) = quiet {
+                            self.quiet = quiet;
+                            changed = true;
+                        }
+
+                        if let Some(output) = output {
+                            self.output = output;
+                            changed = true;
+                        }
```

Comment:
If we wanted to be too picky, then I would say we should check if that's a no-op (new values are equal to the old ones). But that's not really important.

Reply:
I think 65005407e should make you happy though :)

Reply:
Thanks :)